surface/include/pcl/surface/impl/convex_hull.hpp

```diff
@@ -182,106 +166,88 @@ pcl::ConvexHull<PointInT>::performReconstruction2D (PointCloud &hull, std::vecto
   }
    
   // Compute convex hull
-  int exitcode = qh_new_qhull (dimension, static_cast<int> (indices_->size ()), points, ismalloc, const_cast<char*> (flags), outfile, errfile);
-#ifdef HAVE_QHULL_2011
-  if (compute_area_)
+  try
   {
-    qh_prepare_output();
-  }
-#endif
-    
-  // 0 if no error from qhull or it doesn't find any vertices
-  if (exitcode != 0 || qh num_vertices == 0)
-  {
-    PCL_ERROR ("[pcl::%s::performReconstrution2D] ERROR: qhull was unable to compute a convex hull for the given point cloud (%lu)!\n", getClassName ().c_str (), indices_->size ());
-
-    hull.points.resize (0);
-    hull.width = hull.height = 0;
-    polygons.resize (0);
+    orgQhull::Qhull qhull;
+    qhull.runQhull("", dimension, indices_->size (), points.data(), "");
 
-    qh_freeqhull (!qh_ALL);
-    int curlong, totlong;
-    qh_memfreeshort (&curlong, &totlong);
-
-    return;
-  }
-
-  // Qhull returns the area in volume for 2D
-  if (compute_area_)
-  {
-    total_area_ = qh totvol;
-    total_volume_ = 0.0;
-  }
-
-  int num_vertices = qh num_vertices;
-  hull.points.resize (num_vertices);
-  memset (&hull.points[0], static_cast<int> (hull.points.size ()), sizeof (PointInT));
+    // 0 if no error from qhull or it doesn't find any vertices
+    if (qhull.qhullStatus() != 0 || qhull.vertexCount() == 0)
+    {
+      throw std::runtime_error(qhull.qhullMessage());
```

Comment:
Why are we throwing here and not simply returning early and printing the error in the catch block bellow? Is anything else throwing?